executing/executing.py

```diff
@@ -530,6 +539,110 @@ def visit_ClassDef(self, node):
         self.stack.pop()
 
 
+class PositionNodeFinder(object):
+    def __init__(self, frame, stmts, tree, lasti, source):
+        # we can use co_positions() since 3.11, which has fewer limitations
+
+        bc_list = list(dis.Bytecode(frame.f_code, show_caches=True))
+
+        def find_node(index, ignore_positions=(), typ=None):
+            position = bc_list[index // 2].positions
+
+            def match(a, b):
+                for idx, attr in enumerate(
+                    ["lineno", "end_lineno", "col_offset", "end_col_offset"]
+                ):
+
+                    if attr not in ignore_positions and a[idx] != b[idx]:
+                        return False
+
+                return True
+
+
+            return only(
+                node
+                for node in source._nodes_by_line[position[0]]
+                if isinstance(
+                    node, (ast.expr, ast.stmt, ast.excepthandler, ast.MatchAs)
+                )
+                if not isinstance(node, ast.Expr)
+                if typ is None or isinstance(node, typ)
```

Comment:
I think it'd be better to make `(ast.expr, ast.stmt, ast.excepthandler, ast.MatchAs)` the default value of `typ`.

Reply:
good idea 👍